lib/spoom/sorbet/translate_sigs.rb

```diff
@@ -0,0 +1,164 @@
+        def rbi_to_rbs(ruby_contents)
+          ruby_contents = ruby_contents.dup
+
+          tree = RBI::Parser.parse_string(ruby_contents)
+
+          translator = RBI2RBS.new
+          translator.visit(tree)
+          sigs = translator.sigs.sort_by { |sig, _rbs_string| -T.must(sig.loc&.begin_line) }
+
+          sigs.each do |sig, rbs_string|
+            scanner = Scanner.new(ruby_contents, Encoding::UTF_8)
+            start_index = scanner.find_char_position(
+              line: T.must(sig.loc&.begin_line) - 1,
+              character: T.must(sig.loc).begin_column,
+            )
+            end_index = scanner.find_char_position(
+              line: sig.loc&.end_line&.-(1),
+              character: T.must(sig.loc).end_column,
+            )
+            ruby_contents[start_index...end_index] = rbs_string
+          end
+
+          ruby_contents
+        end
+      end
+
+      class RBI2RBS < RBI::Visitor
+        extend T::Sig
+
+        sig { returns(T::Array[[RBI::Sig, String]]) }
+        attr_reader :sigs
+
+        sig { void }
+        def initialize
+          super
+          @sigs = T.let([], T::Array[[RBI::Sig, String]])
+        end
+
+        sig { override.params(node: T.nilable(RBI::Node)).void }
+        def visit(node)
+          return unless node
+
+          case node
+          when RBI::Method
+            translate_method_sigs(node)
+          when RBI::Attr
+            translate_attr_sigs(node)
+          when RBI::Tree
+            visit_all(node.nodes)
+          end
+
+          super
+        end
+
+        private
+
+        sig { params(node: RBI::Method).void }
+        def translate_method_sigs(node)
+          node.sigs.each do |sig|
+            out = StringIO.new
+            p = RBI::RBSPrinter.new(out: out, indent: sig.loc&.begin_column)
+
+            if node.sigs.any?(&:is_abstract)
+              p.printn("# @abstract")
+              p.printt
+            end
+
+            if node.sigs.any?(&:is_override)
+              p.printn("# @override")
+              p.printt
+            end
+
+            if node.sigs.any?(&:is_overridable)
+              p.printn("# @overridable")
+              p.printt
+            end
+
+            p.print("#: ")
+            p.send(:print_method_sig, node, sig)
+
+            @sigs << [sig, out.string]
+          end
+        end
+
+        sig { params(node: RBI::Attr).void }
+        def translate_attr_sigs(node)
+          node.sigs.each do |sig|
+            out = StringIO.new
+            p = RBI::RBSPrinter.new(out: out)
+            p.print_attr_sig(node, sig)
+            @sigs << [sig, "#: #{out.string}"]
+          end
+        end
+      end
+
+      # From https://github.com/Shopify/ruby-lsp/blob/9154bfc6ef/lib/ruby_lsp/document.rb#L127
+      class Scanner
+        extend T::Sig
+
+        LINE_BREAK = T.let(0x0A, Integer)
+
+        # After character 0xFFFF, UTF-16 considers characters to have length 2 and we have to account for that
```

Comment:
I think this complexity wrt to UTF-16 codepoints was only about the fact that VS Code only considers UTF-16 but our code is in UTF-8.

I am not sure if we need the same complexity here. We should be able to treat it as a string of codepoints (or even bytes).